packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,697 @@
+  value="accept"
+  desciption="Accept this API"
+  leading={<ActionListItemIcon icon={StarIcon} />}
+  intent="negative"
+  trailing={<ActionListItemText>⌘ + Space + A</ActionListItemText>}
+  isDefaultSelected={false}
+  onClick={() => {
+    console.log('Item clicked');
+  }}
+/>
+```
+
+Props
+
+```ts
+type ActionListItemProps = {
+  title: string;
+
+  description: string;
+
+  /** Value that we use in JS form. Similar to `value` in `Checkbox` */
+  value: string;
+
+  /**
+   * Leading Icon or assets. Accepts JSX Component
+   *
+   * Should not be allowed when ActionList variant is multiple select
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Trailing Assets. Should ideally be ActionListItemText component for now
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Changes item color to red
+   *
+   * @default neutral
+   */
+  intent?: 'negative' | 'neutral';
+
+  /** Click event on item */
+  onClick?: () => void;
+
+  /**
+   * Set true if you want item to be selected by default
+   *
+   * Should not be allowed when intent is set to negative
+   */
+  isDefaultSelected?: boolean;
+
+  /** URL. Should render item as `a` when this is passed */
+  href?: string;
+};
+```
+
+### `ActionListSection`
+
+Just an API that wraps section inside ActionList.
+
+This entire component is optional. So it may not be added in the ActionList that doesn't have any sections.
+
+It should also add divider after all items.
+
+#### API
+
+```jsx
+<ActionList>
+  <ActionListSection title="Section Heading">
+    <ActionListItem
+    // ...props
+    />
+  </ActionListSection>
+</ActionList>
+```
+
+<details>
+<summary>Previously Considered API</summary>
+
+```jsx
+<ActionList>
+  <ActionListSectionHeading title="Section Heading" />
+  <ActionListItem
+  // ...props
+  />
+  <ActionListDivider />
+</ActionList>
+```
+
+We dropped this because people might tend to put Dividers after every item. It doesn't really explain grouping in any way whereas `ActionListSection` API clearly defines the sections in items.
+
+</details>
+
+#### Props
+
+```ts
+type ActionListSectionProps = {
+  /** Section Heading. Should be optional */
+  title?: string;
+};
+```
+
+### `ActionListHeader`
+
+Header of the ActionList
+
+<img src="2022-11-29-17-36-50.png" alt="ActionListHeader Figma Design" width="300" />
+
+#### Design
+
+- [Figma - ActionListItem / Header](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
+
+#### API
+
+Usage
+
+```jsx
+<ActionListHeader title="Recent Searches" icon={HistoryIcon} />
+```
+
+Props
+
+```ts
+type ActionListHeaderProps = {
+  title: string;
+  icon: IconComponent;
+};
+```
+
+### `ActionListFooter`
+
+> **Warning**
+>
+> API yet to be finalized
+
+Footer of the ActionList
+
+![](2022-11-29-17-40-17.png)
+
+#### Design
+
+- [Figma - ActionListItem / Footer](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
```

Comment:
the action seems to be full width in leading example but with trailing it seems to be a non-full width variant. is that intentional or there's some criteria to switch between full width vs non-fullwidth?

Reply:
It becomes full width when there is not footer title, description or other things.